### PR TITLE
CORS & SPIFFE

### PR DESCRIPTION
**Pull request by casibbald**
> [!NOTE]
> Adds RFC-compliant, route-configurable CORS and a SPIFFE JWT SVID provider (with JWKS verification and token revocation), and hardens JWKS caching/refresh; updates config, examples, tests, and docs.
> 
> - **Security**:
>   - **SPIFFE Provider**: New `SpiffeProvider` with JWT SVID validation, JWKS signature verification, trust-domain/audience checks, leeway, and optional token revocation.
>   - **JWKS**: Stabilize background refresh (atomic claim, TTL change detection, cache/thread fixes), stricter URL validation, and cache/timeout handling.
> - **CORS**:
>   - **RFC-compliant Middleware**: Full CORS implementation with preflight, credentials, exposed headers, max-age; header case/port handling fixes; builder API.
>   - **Route Policies**: `x-cors` (`inherit`/`false`/custom) and per-route merging; global config via `config.yaml`.
> - **Config & Examples**:
>   - Add `cors:` block to `config.yaml` (templates & pet_store) and wire into app startup.
>   - OpenAPI examples annotated with `x-cors`.
> - **Tests**:
>   - Extensive CORS, JWKS, SPIFFE, concurrency, timeout, refresh, and edge-case coverage (unit + integration + benches).
> - **Docs**:
>   - New beginner guide and in-depth blog; README/CONTRIBUTING/TEST docs reorganized to reflect CORS/SPIFFE.
> - **Deps/Infra**:
>   - Bump `reqwest`/`tower-http`; add `tiny_http` and minor supporting crates; CI k8s resource tweaks.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9d34c627efcc28116c1a539790d6c99471a5ac7f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/middleware/cors/mod.rs [hunk 1]
```diff
@@ -0,0 +1,850 @@
...
+            .join(", ");
+        res.set_header("access-control-allow-methods", methods);
+
+        // Set allowed headers
+        let headers = allowed_headers.join(", ");
+        res.set_header("access-control-allow-headers", headers);
+
+        // Add credentials header if enabled
+        if allow_credentials {
+            res.set_header("access-control-allow-credentials", "true".to_string());
```
[the remark this answers is not in the document]
Comment (casibbald) [reply src/middleware/cors/mod.rs]: @copilot  recheck this
